### PR TITLE
Refactor of Board class

### PR DESCRIPTION
**Pull request by jamogriff**
This PR implements refactoring changes to `board.rb` and `board_spec.rb`.

**Larger changes:**
- `generate_board_hash` added as helper method to `initialize` and returns `@cells` when called in initialize
- 3 helper methods created to simplify logic of `valid_placement?`. Associated tests also written to validate their behavior.
   - `player_input_valid?` validates that number of coordinates entered is correct, they exist on the board and are open
   - `is_horizontal?` checks to see if a series of coordinates are horizontal
   - `is_vertical?` checks to see if a series of coordinates are vertical
 
**Smaller changes:**
- All methods that return a boolean value have been renamed to use a question mark at the end. Example: `check_consecutive` now changed to `consecutive?`

Note: further refactoring should examine opportunities to group methods into a new class. Coordinate class?

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -1,11 +1,5 @@
-require_relative 'board'
```
Comment (jamogriff) [inline lib/board.rb:1]: Don't know why this commit is on this branch, when I committed and pushed to #16...

File: spec/board_spec.rb [hunk 2]
```diff
@@ -65,27 +66,62 @@
...
+      expect(board.player_input_valid?(sub, ["B1", "B2"])).to eq(false)
+    end
+  end
+
+  describe "#is_vertical? and #is_horizontal?" do
+    board = Board.new
+    cruiser = Ship.new("Cruiser", 3)
+
+    it 'returns true if ship is horizontal' do
+      expect(board.is_horizontal?(cruiser, ["A1", "B1", "C1"]))
```
Comment (aliyamerali) [inline spec/board_spec.rb]: add .to eq(true)

File: spec/board_spec.rb [hunk 3]
```diff
@@ -65,27 +66,62 @@
...
+    end
+
+    it 'returns true is ship is vertical' do
+      expect(board.is_vertical?(cruiser, ["B2", "B3", "B4"]))
```
Comment (aliyamerali) [inline spec/board_spec.rb]: add .to eq(true)

**Review by aliyamerali — CHANGES_REQUESTED** [review]
Finish out the last two tests assertion statements please :)

**Comment by jamogriff** [comment]
I made those changes under 530b99ee5e32d8a26eee11b7e52d52466cf2f125

**Review by aliyamerali — APPROVED** [review]

**Pull request merged**